### ISSUE TITLE
bug: `emulate`is not used for E2E Screenshot tests

### ISSUE THREAD

**Issue by mlechler** [ionic-team/stencil#5353]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

4.12.1

### Current Behavior

We have four viewports defined in the `emulate` array in the `stencil.config.ts`. When running our e2e screenshot tests these defined viewports aren't used. As mentioned [here](https://github.com/ionic-team/stencil/issues/4601#issuecomment-1648447341), I understood, that the `emulate` array should still be used for screenshot tests, which isn't the case.

### Expected Behavior

Use the `emulate` array of the `stencil.config.ts` or provide a easy way to use multiple viewports for every e2e screenshot test, without setting every single one of it.

### System Info

```shell
System: node 20.11.0
    Platform: darwin (21.6.0)
   CPU Model: Apple M1 Pro (10 cpus)
    Compiler: /node_modules/@stencil/core/compiler/stencil.js
       Build: 1707148558
     Stencil: 4.12.1 🏸
  TypeScript: 5.3.3
      Rollup: 2.56.3
      Parse5: 7.1.2
      jQuery: 4.0.0-pre
      Terser: 5.27.0
```

### Steps to Reproduce

Run `npm run test.screenshot` in the test repo.
Should generate screenshot with 600x600, but generates 800x600

### Code Reproduction URL

https://github.com/mlechler/stencil-e2e-test-bug

**Comment by tanner-reits** [comment]
@mlechler I confirmed that the generated screenshots do not have the same dimensions as what is being set in the Stencil config. I'll get this ingested in our backlog for someone to take a look at.

**Comment by mlechler** [comment]
Hi @tanner-reits, what is the time horizon on this issue?

**Comment by tanner-reits** [comment]
@mlechler I don't have a definitive timeline right now. I spent some more time playing around with a fix, but we uncovered some additional discrepancies in the generated output. The team needs to dig in deeper to the screenshot testing process so we're confident in the change(s) we make. This is an experimental feature in Stencil after all, so there are many kinks and nuances we need to work through.

**Comment by christian-bromann** [comment]
@mlechler thanks to @tanner-reits work we were able to publish a dev release with a fix for this bug. You can install it in your project via:

```sh
npm i @stencil/core@4.12.2-dev.1709767698.0a1aa6b
```

Can you verify that this resolves this issue?

**Comment by christian-bromann** [comment]
> the viewport works great now! 🥳

Awesome, I will go ahead and close the issue.

> Is this an additional bug or is it intended to work like that? If it is intended: How can we emulate different themes (color & spacing) in our screenshot tests?

Mind raising a new issue for this?

**Comment by mlechler** [comment]
Great thanks @christian-bromann!

I opened #5459 for the new issue.

**Comment by christian-bromann** [comment]
The fix has been released with Stencil 🚞 [v4.13.0](https://github.com/ionic-team/stencil/releases/tag/v4.13.0)
